### ISSUE TITLE
Create a 'steps to try before submitting a bug' checklist

### ISSUE THREAD

**Issue by 0xazure** [new-xkit/XKit#230]
We've had a number of repeat issues, some kind of checklist for users affected by bugs to try before submitting a report would help cut down on answering these kinds of issues.

I'd like to see the list on the new-xkit-extension ask/submit page, and @bvtsang suggested having it somewhere in XKit itself.  Maybe an embedded gist, so it's easy to update?

**Comment by Wolvan** [comment]
either a gist or a pastebin to embbed into XKit sounds like a good idea. Can you get a raw gist?

**Comment by 0xazure** [comment]
Yup.  After creating a gist there's a button to get the raw content.  You can get gist content pinned to a version, or (with some URI ~~deletion~~ manipulation) just the latest.

**Comment by Wolvan** [comment]
I guess that sounds good enough. Will we just list all issues there or solutions or something the like?

**Comment by 0xazure** [comment]
I figure 'common resolutions'.  Some examples:

- Make sure XKit is at the latest version
  - Need to include steps to manually update extensions on all supported browsers
- Make sure your browser is up to date
  - Steps to update browsers
- If your browser is old, errors, and won't update, reinstall your browser
- Try disabling XKit (if it's not directly XKit-related, like a problem with one of our extensions) and see if it's a Tumblr change

etc.

**Comment by Wolvan** [comment]
That sounds good to me, how about we put these common solutions into XKit Bridge though, as they are extremely browser dependent. Just a string or html variable that we can later print to the user.
And do we already have checks for outdated browsers in place?

**Comment by 0xazure** [comment]
I still think putting it into a gist or integrated into our `gh-pages` releases would be best.  We can display browser-relevant solutions with a little logic (we already test for browser type), but putting strings into the `bridge.js` code would tie it to slow browser extension releases.

As for testing browser versions, I have no idea.  I doubt we do though.

**Comment by bvtsang** [comment]
Draft steps to try before submitting a bug:

1. Make sure you have the latest version of your browser.
  - To update Chrome, click the hamburger menu > About Google Chrome
  - To update Firefox, click the hamburger menu > Open Help Menu (help icon) > About Firefox
2. Make sure you have the latest version of the new XKit.
  - To view your version of XKit, click on the XKit button > About + Support tab
  - To force XKit to update, click on the XKit button > Other > Update All > Update all my extensions
3. Make sure the bug hasn't been filed before.
  - The list of known issues is available [here](https://github.com/new-xkit/XKit/issues)
  - You can use the search bar to find your issue or filter by label to help you narrow down on a problem with a particular browser or extension.
4. If you can't find the bug in our list of known issues, you can file a new one.
  - If you don't yet have a Github account, please register for one first.
  - Visit [our issues page](https://github.com/new-xkit/XKit/issues) and click the New Issue button
  - When describing your issue, be sure to include steps to reproduce the bug. We need to be able to confirm the bug before we can fix it.
5. After filing the issue, check it periodically for updates. Alternatively, [enable notifications from Github](https://github.com/settings/notifications) to receive emails as activity happens.

**Comment by 0xazure** [comment]
After some discussion on Gitter, we've decided against hosting it as a gist and instead want an analogue to the `CONTRIBUTING.md`, `README.md` files at the top level of the project.  We've decided on `TROUBLESHOOTING.md` for the filename, we just have to populate it with the steps.
